### DISCUSSION TITLE
Arrays displayed as [object, Object]

### DISCUSSION THREAD

**Discussion by victorlmneves** [histoire-dev/histoire#182] [Q&A] [answered]
I'm wondering why the arrays are displayed as [object, Object] while objects are displayed correctly
From your examples
https://vue3.examples.histoire.dev/story/src-components-codegen-story-vue?variantId=props
![image](https://user-images.githubusercontent.com/17341789/179497895-a1f60293-463a-4c0e-8f53-2d4992ecc99c.png)

On a story that I'm working on
![image](https://user-images.githubusercontent.com/17341789/179498166-70fcd949-a985-4907-8f71-4fb655dba98d.png)

Thanks

**Comment by victorlmneves** [comment]
Now I understood the reason for rendering [object Object]
It's related to the input being a `textarea` and saw that you are working on a new element for rendering JSON
[https://github.com/histoire-dev/histoire/issues/30]

**Comment by Akryum** [comment, marked as answer]
You can make your own control and display it in the `#controls` slots :wink:

**Reply by victorlmneves** [reply]
Do you mean to create the JSON builtin control as you already have for input, textarea, radio, checkboxes, and so on?

**Reply by Akryum** [reply]
You can put any component you want in the controls. So you could totally create some components to edit your story data :)

**Reply by victorlmneves** [reply]
Going to try to see if I can make it work 😅

**Reply by victorlmneves** [reply]
@Akryum It's done
Here's the code for someone else who needs to implement
```
<script setup lang="ts">
import { computed } from 'vue';

const props = withDefaults(
    defineProps<{
        title?: string;
        modelValue: {
            [key: string]: string | unknown;
        };
    }>(),
    { title: 'JSON' }
);
const emit = defineEmits(['update:modelValue']);
const updateValue = (value) => {
    emit('update:modelValue', value);
};
const mutatedJSON = computed({
    get() {
        return JSON.stringify(props.modelValue, null, 4);
    },

    set: (value) => {
        try {
            updateValue(JSON.parse(value));
        } catch (e) {
            return;
        }
    },
});
</script>

<template>
    <HstTextarea
        v-model="mutatedJSON"
        :title="title"
        placeholder="Enter JSON"
    ></HstTextarea>
</template>

```
